### ISSUE TITLE
Add "DocsLink" feature

### ISSUE THREAD

**Issue by PropGit** [parallaxinc/Propeller-Tool#45]
Twyyx (forum) during 11/18/2020 P2 Live Forum suggested making a "DocsLink" feature where some action on a word in source (such as CTRL+Double-Click) causes Propeller Tool to emit a URL POST (in the form of a user-configurable domain plus the Propeller type plus the selected word/phrase.  Developers can make use of that to provide documentation or other info based on those keywords.

From Twyyx on the forum:

Jeff, if you do add the docslink feature we talked about, I'd like to know even if you don't publish it as a user-facing feature.
I would like to do some R&D and experimentation for linking in some docs on an editable view. If I get anything of value to come from it, I'll share it back and demo to you and whoever is interested in learning about the idea.
I can't guarantee success, but I have some particular ideas on how to streamline docs construction, feedback, and hosting by leveraging some specific projects already out there.

**Comment by PropGit** [comment]
@jshook - I've been considering this more.  

The easiest is to implement it the way you suggested.  I also dreamed up some enhancements from there that can be implemented later if deemed helpful.

- Make the DocsLink action bring up a small overlay near the cursor to show summarized information, with a button to read more.  This would require that the "service" provide a result like that, perhaps via a simple REST interface, or it could fetch the HTML and simply pare it down to a limited summary.
  - This could act like a very quick, in-app informative response that gives the user everything needed for a "refresher" or "intro" and then it's out of the way again, whether or not the user chooses the _read more_ option.
- I think it'd be best if it's not an exclusive implementation - meaning that it'd be great if multiple sources could be entered and the DocsLink activation could provide a summary from each (for further exploration) or some other way to make each source visible. 

Thoughts?

**Comment by jshook** [comment]
I was thinking in this direction as well. However, I did avoid elaborating too much because I knew I was asking for something that could be a distraction while you are all so busy. The limited version of this is good enough to get started with and also enable those who want to prototype doc site support to work with it.

That said, I think we're both imagining that we can do much more, and I like the direction of it. The local views, multiple sources, summary vs details.. it's all nice to have.

On local views, to keep the implementation simple, it would be best to call out to a "summary" if that is what is wanted, and also have a different view which is "detail". This would allow the tool to just look at HTTP and just display the contents in a borrowed browser frame. attached to the proptool (I'm not clear on how that works, but it seem to be a common tactic.)

For the calling interface to be simple, you can provide a couple of tokens in addition to the ones we already provided. It would be up to the person configuring the URL what the target scheme is. For example, you can be selective with 'htts://my.docs.example.com/$PASMVER$/$LEVEL_OF_DETAIL$/$TERM$.html` where each of the tokens is required to be one of a known set. LEVEL_OF_DETAIL could be required to always be either 'summary' or 'detail' or 'see_also'.

On multiple sources, I like the idea, but it seems to fit more in the realm of "see_also" as hinted above. This is because you will prop tool users to be able to suggest improvements to a common set of community docs, or at least docs which can be used as the standard source. We just don't have one that fits this form yet. I wouldn't expect users to be reconfiguring their docs URL in general, and if we allowed it, there should be a really simple way to let them reset back to the "standard docs", and also a way to warn them if they've changed this setting and their doc lookups come back empty or with errors.

**Comment by PropGit** [comment]
Yes, on the reset option, the preference window keeps defaults available and a simple click of the Reset (or is it Restore?) option will reset them back to defaults.

Thanks for the feedback.  If you have more ideas, or more details you want to provide, put them here anytime.  

My thoughts are that initially, the extra source(s) will be very beneficial - then over time it may naturally culminate into just one source (or maybe two) that people use.  Ideally, we'd like to make our documentation improvable in real time (or near real time) by customers, with the whole set of it online and consumable in bite-sized amounts.  Tutorials could be written blog format, with links to the deeper bite-sized tidbits sprinkled in for good measure.
